src/utilities/common.tsx

```diff
@@ -203,3 +204,26 @@ export function formatClassNames(className: string, colon: boolean = false, layo
   const hasColon = colon && layout !== LAYOUT_TYPES.VERTICAL;
   return cx(`${className}-${layout}`, `${className}${hasColon ? '' : '-no'}-colon`);
 }
+
+export const unflattenObject = (object: Object) => {
+  const flattenedObject = flatten(object, { removeDate: true }),
+    periodsAndBracketsRegex = /[.[\]]/,
+    result = {};
+
+  Object.keys(flattenedObject).forEach((key: string) => {
+    const keys = key.split(periodsAndBracketsRegex).filter((key: string) => key !== '');
+
+    keys.reduce(
+      (accumulator: any, current: string, index: number) =>
+        accumulator[current] ||
+        (accumulator[current] = Number.isNaN(Number(keys[index + 1]))
+          ? keys.length - 1 === index
+            ? flattenedObject[key]
+            : {}
+          : []),
+      result,
+    );
+  });
+
+  return result;
```

Comment:
This is a truly impressive piece of code. However, I think we can remove most of this logic by using `_.set` to handle the array and dot notation to object placement for us:

```ts
  const flattenedObject = flatten(object, { removeDate: true });

  return Object.entries(flattenedObject).reduce(
    (objOut, [key, value]) => set(objOut, key, value), {}
 );
```

Reply:
Thanks! but I had some help from good 'ol stackOverflow with this one. And yeah I always forget about lodash will update